saleor/tests/e2e/checkout/test_checkout_create_from_order.py

```diff
@@ -0,0 +1,162 @@
+    )
+    shipping_method_id = shipping_method_data["id"]
+
+    create_shipping_method_channel_listing(
+        e2e_staff_api_client, shipping_method_id, channel_id
+    )
+
+    product_type_data = create_product_type(
+        e2e_staff_api_client,
+    )
+    product_type_id = product_type_data["id"]
+
+    category_data = create_category(e2e_staff_api_client)
+    category_id = category_data["id"]
+
+    product_data = create_product(e2e_staff_api_client, product_type_id, category_id)
+    product_id = product_data["id"]
+
+    create_product_channel_listing(e2e_staff_api_client, product_id, channel_id)
+
+    stocks = [
+        {
+            "warehouse": warehouse_id,
+            "quantity": 5,
+        }
+    ]
+    product_variant_data = create_product_variant(
+        e2e_staff_api_client,
+        product_id,
+        stocks=stocks,
+    )
+    product_variant_id = product_variant_data["id"]
+
+    create_product_variant_channel_listing(
+        e2e_staff_api_client,
+        product_variant_id,
+        channel_id,
+    )
+    data = draft_order_create(
+        e2e_staff_api_client,
+        channel_id,
+    )
+
+    order_id = data["order"]["id"]
+    order_lines = [{"variantId": product_variant_id, "quantity": 1, "price": 100}]
+    order_data = order_lines_create(e2e_staff_api_client, order_id, order_lines)
+    order_product_variant_id = order_data["order"]["lines"][0]["variant"]
+    order_product_quantity = order_data["order"]["lines"][0]["quantity"]
+
+    return (
+        order_id,
+        order_product_variant_id,
+        order_product_quantity,
+    )
+
+
+@pytest.mark.e2e
+def test_checkout_create_from_order_core_0104(
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_product_types_and_attributes,
+    permission_manage_shipping,
+    permission_manage_orders,
+    permission_manage_checkouts,
+):
+    # Before
+    (
+        order_id,
+        order_product_variant_id,
+        order_product_quantity,
+    ) = prepare_product(
+        e2e_staff_api_client,
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_product_types_and_attributes,
+        permission_manage_shipping,
+        permission_manage_orders,
+        permission_manage_checkouts,
+        channel_slug="test",
+    )
+
+    # Step 1 - Create checkout from order
+
+    checkout_data = checkout_create_from_order(e2e_staff_api_client, order_id)
+    checkout_id = checkout_data["checkout"]["id"]
+    assert checkout_id is not None
+    errors = checkout_data["errors"]
+    assert errors == []
+    checkout_lines = checkout_data["checkout"]["lines"]
+    assert checkout_lines != []
+
+    checkout_product_variant_id = checkout_lines[0]["variant"]["id"]
+    checkout_product_quantity = checkout_lines[0]["quantity"]
+    assert checkout_product_variant_id is not None
+    assert checkout_product_quantity is not None
```

Comment:
should we double check it since it is also compared in lines 161, 162?

Reply:
@Air-t You are right - I've removed these asertions https://github.com/saleor/saleor/pull/13509/commits/c802020e05d8d2daf984fc9e36476f0f057889e4